src/controls/private/QskLevelingSensor.cpp

```diff
@@ -0,0 +1,178 @@
+#include <QskFunctions.h>
+#include <QskLevelingSensor.h>
+#include <QskScaleTickmarks.h>
+
+#include <type_traits>
+#include <unordered_map>
+
+#include <QVector2D>
+#include <QVector3D>
+
+namespace
+{
+    template< typename T >
+    bool compareExchange( T& dst, const T& src )
+    {
+        if ( dst != src )
+        {
+            dst = src;
+            return true;
+        }
+        return false;
+    }
+
+    template<>
+    bool compareExchange< float >( float& dst, const float& src )
+    {
+        if ( !qskFuzzyCompare( dst, src ) )
+        {
+            dst = src;
+            return true;
+        }
+        return false;
+    }
+
+    template<>
+    bool compareExchange< double >( double& dst, const double& src )
+    {
+        if ( !qskFuzzyCompare( dst, src ) )
+        {
+            dst = src;
+            return true;
+        }
+        return false;
+    }
+
+    inline bool isAxis( const Qt::Axis axis )
+    {
+        return axis == Qt::XAxis || axis == Qt::YAxis || axis == Qt::ZAxis;
+    }
+}
+
+QSK_SUBCONTROL( QskLevelingSensor, OuterDisk )
+QSK_SUBCONTROL( QskLevelingSensor, Horizon )
+QSK_SUBCONTROL( QskLevelingSensor, TickmarksX )
+QSK_SUBCONTROL( QskLevelingSensor, TickmarksXLabels )
+QSK_SUBCONTROL( QskLevelingSensor, TickmarksY )
+QSK_SUBCONTROL( QskLevelingSensor, TickmarksYLabels )
+QSK_SUBCONTROL( QskLevelingSensor, TickmarksZ )
+QSK_SUBCONTROL( QskLevelingSensor, TickmarksZLabels )
+
+#define RETURN_IF_FALSE( expr )                                                                    \
+    if ( !( expr ) )                                                                               \
+        return;
+
+class QskLevelingSensor::PrivateData
+{
+  public:
+    QVector3D m_angle = { 45, 45, 45 };
+    Tickmarks m_tickmarks[ 3 ];
+    TickmarksLabels m_tickmarksLabels[ 3 ];
+    std::unordered_map< QskAspect::Subcontrol, QVector3D > m_subControlRotation;
+};
+
+QskLevelingSensor::QskLevelingSensor( QQuickItem* const parent )
+    : Inherited( parent )
+    , m_data( std::make_unique< QskLevelingSensor::PrivateData >() )
```

Comment:
What is the benefit of using std::make_unique here ?

If there is one we should change all the other classes, where the PrivateData pointer is initialized by a simple new.